cpp/src/arrow/flight/sql/CMakeLists.txt

```diff
@@ -117,18 +119,39 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
     endif()
   endif()
 
+  # Fetch ADBC test suite
+  cmake_minimum_required(VERSION 3.11)
+  include(FetchContent)
+  fetchcontent_declare(adbc
+                       GIT_REPOSITORY https://github.com/apache/arrow-adbc.git
+                       GIT_TAG ddd715a443cf2e5edd15c639af3845afe81af4ee)
+  fetchcontent_makeavailable(adbc)
```

Comment:
@kou What do you think about this?

Reply:
How about creating `adbc_validation` package for https://github.com/apache/arrow-adbc/tree/main/c/validation in apache/arrow-adbc and using installed `adbc_validation` package from apache/arrow instead of using `FetchContent`?

Reply:
Just to make sure, do you mean a Conda package or similar?

Reply:
Hmm, no. I meant the following:

1. Create `CMakeLists.txt` in https://github.com/apache/arrow-adbc/tree/main/c/validation like https://github.com/apache/arrow-adbc/blob/main/c/driver_manager/CMakeLists.txt
2. Apache Arrow C++ Flight SQL developers install https://github.com/apache/arrow-adbc/tree/main/c/validation by 1.
3. Apache Arrow C++ uses 2.

We may also implement 2. in `cpp/cmake_modules/ThirdpartyToolchain.cmake` like other dependency.

Reply:
I see. I will do that.

Reply:
And now it points at the main branch again.